src/robotFactory.js

```diff
@@ -1,10 +1,90 @@
 'use strict';
 
-class BaseRobot {}
+const BASIC_COORDS_X_Y = {
+  x: 0,
+  y: 0,
+};
+
+const BASIC_COORDS_X_Y_Z = {
+  x: 0,
+  y: 0,
+  z: 0,
+};
+
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+
+    this.coords = {
+      ...BASIC_COORDS_X_Y,
+      ...coords,
+    };
+    this.chipVersion = chipVersion;
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, `
+      + `Chip version: ${this.chipVersion}, `
+      + `Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    this.coords = {
+      ...BASIC_COORDS_X_Y_Z,
+      ...coords,
+    };
```

Comment:
```suggestion
    this.coords.z = coords.z || 0;
```
There is no need to duplicate `x` and `y`. You can add only a new coord